onnx/serialization.py

```diff
@@ -65,11 +68,26 @@ def get(self, fmt: str) -> ProtoSerializer:
                 f"Unsupported format: '{fmt}'. Supported formats are: {self._serializers.keys()}"
             ) from None
 
+    def get_format_from_file_extension(self, file_extension: str) -> str | None:
+        """Get the corresponding format from a file extension.
+
+        Args:
+            file_extension: The file extension to get a format for.
+
+        Returns:
+            The format for the file extension, or None if not found.
+        """
+        for fmt, serializer in self._serializers.items():
```

Comment:
Maybe it doesn't matter much, but I wonder if this should be maintained as a map to avoid iterating everytime. (Of course, it could be a composed map to go from extension to the serializer, making it even more direct.)

Reply:
Returning the format and not the serializer makes the resolution logic more uniform: https://github.com/onnx/onnx/pull/5347/files/6334d6dedcf37738f52485ee028fa28cbcaeeb4b#diff-ae94ce8411086ae2f1159a84ee69aa79c2acddffbc2251d543bef7732aac2e78R176-R184

At the time I decided to not use a map because I didn't want to maintain one. Thinking again maybe it makes sense when we do not allow un-registering a serializer. If we do the maintenance of the mapping will be more complicated.

Reply:
Done